var/spack/repos/builtin/packages/py-keras/package.py

```diff
@@ -40,13 +46,86 @@ class PyKeras(PythonPackage):
+    depends_on('py-tensorflow@2.7.0:2.7', type=('build', 'run'), when='@2.7.0:2.7')
+    depends_on('bazel', type='build', when='@2.5.0:')
+    depends_on('protobuf', type='build', when='@2.5.0:')
+
+    def url_for_version(self, version):
+        if version >= Version('2.6.0'):
+            url = 'https://github.com/keras-team/keras/archive/refs/tags/v{0}.tar.gz'
+            return url.format(version.dotted)
+        else:
+            return super(PyKeras, self).url_for_version(version)
+
+    @when('@2.5.0:')
+    def patch(self):
+        infile = join_path(self.package_dir, 'protobuf_build.patch')
+        with open(infile, 'r') as source_file:
+            text = source_file.read()
+        with open('keras/keras.bzl', mode='a') as f:
+            f.write(text)
+
+        filter_file('load("@com_google_protobuf//:protobuf.bzl", "py_proto_library")',
+                    'load("@org_keras//keras:keras.bzl", "py_proto_library")',
+                    'keras/protobuf/BUILD',
+                    string=True)
+
+    @when('@2.5.0:')
+    def build(self, spec, prefix):
+        self.tmp_path = tempfile.mkdtemp(dir='/tmp', prefix='spack')
+        env['TEST_TMPDIR'] = self.tmp_path
+        env['HOME'] = self.tmp_path
+
+        args = [
+            # Don't allow user or system .bazelrc to override build settings
+            '--nohome_rc',
+            '--nosystem_rc',
+            # Bazel does not work properly on NFS, switch to /tmp
+            '--output_user_root=' + self.tmp_path,
+            'build',
+            # Spack logs don't handle colored output well
+            '--color=no',
+            '--jobs={0}'.format(make_jobs),
+            # Enable verbose output for failures
+            '--verbose_failures',
+            # Show (formatted) subcommands being executed
+            '--subcommands=pretty_print',
+            '--spawn_strategy=local',
+            # Ask bazel to explain what it's up to
+            # Needs a filename as argument
+            '--explain=explainlogfile.txt',
+            # Increase verbosity of explanation,
+            '--verbose_explanations',
+            # bazel uses system PYTHONPATH instead of spack paths
+            '--action_env', 'PYTHONPATH={0}'.format(env['PYTHONPATH']),
+            '//keras/tools/pip_package:build_pip_package',
+        ]
+
+        bazel(*args)
+
+        build_pip_package = Executable(
+            'bazel-bin/keras/tools/pip_package/build_pip_package')
+        buildpath = join_path(self.stage.source_path, 'spack-build')
+        build_pip_package('--src', buildpath)
+
+    @when('@2.5.0:')
+    def install(self, spec, prefix):
+        tmp_path = env['TEST_TMPDIR']
+        buildpath = join_path(self.stage.source_path, 'spack-build')
+        with working_dir(buildpath):
+
+            setup_py('install', '--prefix={0}'.format(prefix),
+                     '--single-version-externally-managed', '--root=/')
```

Comment:
See the other TF packages I updated in #27798, this should now be something like:
```suggestion
            args = std_pip_args + ['--prefix=' + prefix, '.']                            
            pip(*args)
```